### PR TITLE
[RFR] Add tabs prop to <TabbedForm> to allow injecting custom Tabs component

### LINKED ISSUE #3116: Scrollable tabs on TabForm
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
I want the possibility to scroll on <Tabs> of <TabbedForm>, when using mobile devices and there are many tabs, some become out of view and is not possible to scroll horizontally to see the hidden ones.

**Additional context**
Material-Ui <Tabs> component receives prop called scrollable that enables horizontal scrolling <TabbedForm> could receive the same prop.

This is similar to https://github.com/marmelab/react-admin/issues/2317 but for TabbedForm.

### PR DESCRIPTION
**Pull request by sagarbakhtar**
- [x] New component TabbedFormTabs to be default prop 'tabs' and replace Tabs in TabbedForm
- [x] Test cases for a new component
- [x] Relevant doc changes

Fixes #3116

### CODE REVIEW

File: docs/CreateEdit.md [hunk 1]
```diff
@@ -427,6 +427,27 @@ To style the tabs, the `<FormTab>` component accepts two props:
 - `className` is passed to the tab *header*
 - `contentClassName` is passed to the tab *content*
 
+### TabbedFormTabs
+By default `<TabbedForm>` uses, `<TabbedFormTabs>` internal react-admin component to renders tabs. You can pass custom component as the `tabs` prop to override the default component. Also props from `<TabbedFormTabs>` are passed to material-ui's `<Tabs>` component inside `<TabbedFormTabs>`.
```
Comment (djhi) [inline docs/CreateEdit.md]:
```suggestion
By default `<TabbedForm>` uses, `<TabbedFormTabs>` internal react-admin component to renders tabs. You can pass a custom component as the `tabs` prop to override the default component. Besides, props from `<TabbedFormTabs>` are passed to material-ui's `<Tabs>` component inside `<TabbedFormTabs>`.
```

File: docs/CreateEdit.md [hunk 2]
```diff
@@ -427,6 +427,27 @@ To style the tabs, the `<FormTab>` component accepts two props:
...
+
+Following example shows how to make use to scrollable `<Tabs>`, you can pass scrollable prop to `<TabbedFormTabs>` and pass that as `tabs` prop to `<TabbedForm>`
```
Comment (djhi) [inline docs/CreateEdit.md]:
```suggestion
Following example shows how to make use of scrollable `<Tabs>`. Pass the `scrollable` prop to `<TabbedFormTabs>` and pass that as the `tabs` prop to `<TabbedForm>`
```

**Comment by sagarbakhtar** [comment]
@djhi Thanks !!

**Comment by djhi** [comment]
Thanks for working on this :) I'll wait for @fzaninotto though as this is technically a new feature and we're working on the `3.0` which includes #3246.

It means we'll probably ask you to target the `next` branch and to ensure the `tabs` prop accepts a component instead of an element.

**Comment by fzaninotto** [comment]
Indeed, please rebase on `next` as it's a new feature.

**Comment by fzaninotto** [comment]
Also, material-ui has implemented scrollable fullwidth tabs in https://github.com/mui-org/material-ui/pull/15670

**Comment by fzaninotto** [comment]
e2e tests seem to fail, can you take a look?

**Comment by sagarbakhtar** [comment]
Yes, I will check it.

**Comment by sagarbakhtar** [comment]
@fzaninotto fixed the issue.

File: docs/CreateEdit.md [hunk 3]
```diff
@@ -427,6 +427,27 @@ To style the tabs, the `<FormTab>` component accepts two props:
 - `className` is passed to the tab *header*
 - `contentClassName` is passed to the tab *content*
 
+### TabbedFormTabs
+By default `<TabbedForm>` uses, `<TabbedFormTabs>` internal react-admin component to renders tabs. You can pass a custom component as the `tabs` prop to override the default component. Besides, props from `<TabbedFormTabs>` are passed to material-ui's `<Tabs>` component inside `<TabbedFormTabs>`.
```
Comment (djhi) [inline docs/CreateEdit.md]:
```suggestion
By default `<TabbedForm>` uses `<TabbedFormTabs>`, an internal react-admin component to renders tabs. You can pass a custom component as the `tabs` prop to override the default component. Besides, props from `<TabbedFormTabs>` are passed to material-ui's `<Tabs>` component inside `<TabbedFormTabs>`.
```

File: docs/CreateEdit.md [hunk 4]
```diff
@@ -427,6 +427,27 @@ To style the tabs, the `<FormTab>` component accepts two props:
...
+
+Following example shows how to make use of scrollable `<Tabs>`. Pass the `scrollable` prop to `<TabbedFormTabs>` and pass that as the `tabs` prop to `<TabbedForm>`
```
Comment (djhi) [inline docs/CreateEdit.md]:
```suggestion
The following example shows how to make use of scrollable `<Tabs>`. Pass the `scrollable` prop to `<TabbedFormTabs>` and pass that as the `tabs` prop to `<TabbedForm>`
```

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**